### PR TITLE
 Improve jvar map interface

### LINKED ISSUE #13951: Improve JvarMapInterface with per parameter maps
## Reason
JvarMapInterface provides a fast method to find a mapping from a `jvar` (MooseVariable number) value to the index _i_ into the coupled variables vector in a kernel or boundary condition.
This simplifies jacobian computations with lots of coupled derivative material properties stored in vectors and indexed by _i_.

I need a generalization of this interface to index not just into the vector of all coupled variables, but into specific couple variable vectors (as defined by the addCoupledVar input parameters).

## Design
A `getJvarMap` method, called in the constructor with an couple variable parameter name as an argument, will build and return a map indexing all non-linear variable numbers that occur in the given coupled variable vector to their respective index in said vector.

## Impact
Improvement

### PR DESCRIPTION
**Pull request by dschwen**
Closes #13951

### CODE REVIEW

File: test/tests/interfaces/jvarmap/tests
Comment (permcody) [inline test/tests/interfaces/jvarmap/tests]: buyild -> build

**Review by permcody — CHANGES_REQUESTED** [review]
Typo!

File: test/tests/interfaces/jvarmap/tests
Comment (dschwen) [reply test/tests/interfaces/jvarmap/tests]: Noooooooooooooooooo!

**Review by permcody — APPROVED** [review]

**Comment by dschwen** [comment]
<sub>preeeessss theeeee buuuutooon...</sub>

**Comment by permcody** [comment]
You can always merge an approved PR even if it’s your own. Also you have merge rights in general but I’ll hit the button for you.

**Comment by dschwen** [comment]
Ok, good to know. Thanks!

**Pull request merged**
